src/common/cmdparse.cc

```diff
@@ -65,6 +65,55 @@ arg_desc_t cmddesc_get_args(const String& cmddesc)
   return arg_desc;
 }
 
+std::string cmddesc_get_prenautilus_compat(const std::string &cmddesc)
+{
+  std::vector<std::string> out;
+  stringstream ss(cmddesc);
+  std::string word;
+  bool changed = false;
+  while (std::getline(ss, word, ' ')) {
+    // if no , or =, must be a plain word to put out
+    if (word.find_first_of(",=") == string::npos) {
+      out.push_back(word);
+      continue;
+    }
+    auto desckv = cmddesc_get_args(word);
+    auto j = desckv.find("type");
+    if (j != desckv.end() && j->second == "CephBool") {
+      // Instruct legacy clients to send --foo-bar string in place
+      // of a 'true'/'false' value
```

Comment:
Can you explain what's happening here a bit more? The whole point of the local_mon_commands versus leader_mon_commands is that the peons are supposed to present a unified command view and forward anything they don't understand to the leader. Did we switch the syntax enough on these options that the pre-nautilus monitors just can't pass them through validation?

Reply:
update